diff --git a/DDCore/src/DetectorImp.cpp b/DDCore/src/DetectorImp.cppindex 12262bb7..c938ad6a 100644
--- a/DDCore/src/DetectorImp.cpp
+++ b/DDCore/src/DetectorImp.cpp
@@ -631,17 +631,22 @@ vector<DetElement> DetectorImp::detectors(const string& type1,
 }

 Handle<NamedObject> DetectorImp::getRefChild(const HandleMap& e, const string& name, bool do_throw) const {
-  HandleMap::const_iterator i = e.find(name);
-  if (i != e.end()) {
-    return (*i).second;
+  HandleMap::const_iterator it = e.find(name);
+  if (it != e.end()) {
+    return it->second;
   }
   if (do_throw) {
-    int cnt = 0;
-    cout << "GetRefChild: Failed to find child with name: " << name-         << " Map contains " << e.size() << " elements." << endl;
-    for(i=e.begin(); i!=e.end(); ++i)
-      cout << "   " << cnt << "  " << (*i).first << endl;
-    throw runtime_error("Cannot find a child with the reference name:" + name);
+    std::stringstream err;
+    err << "getRefChild: Failed to find child with name: " << name
+        << " Map contains " << e.size() << " elements: {";
+    for (it = e.begin(); it != e.end(); ++it) {
+      if (it != e.begin()) {
+        err << ", " << endl;
+      }
+      err << it->first << endl;
+    }
+    err << "}";
+    throw runtime_error(err.str());
   }
   return 0;
 }
