diff --git a/DDG4/hepmc/HepMC3EventReader.cpp b/DDG4/hepmc/HepMC3EventReader.cpp
index 74425aff..88f33b73 100644
--- a/DDG4/hepmc/HepMC3EventReader.cpp
+++ b/DDG4/hepmc/HepMC3EventReader.cpp
@@ -109,6 +109,7 @@ HEPMC3EventReader::readParticles(int event_number, Vertices& vertices, Particles
     p->vex          = vex.get_component(0) * len_unit;
     p->vey          = vex.get_component(1) * len_unit;
     p->vez          = vex.get_component(2) * len_unit;
+    p->vet          = vex.get_component(3) * len_unit / CLHEP::c_light;
     p->process      = 0;
     p->spin[0]      = spin[0];
     p->spin[1]      = spin[1];
@@ -146,7 +147,7 @@ HEPMC3EventReader::readParticles(int event_number, Vertices& vertices, Particles
       vtx->x = p->vex;
       vtx->y = p->vey;
       vtx->z = p->vez;
-      vtx->time = p->time;
+      vtx->time = p->vet;
 
       vtx->out.insert(p->id) ;
     }
diff --git a/DDG4/include/DDG4/Geant4Particle.h b/DDG4/include/DDG4/Geant4Particle.h
index 2c3cb717..ca8711a7 100644
--- a/DDG4/include/DDG4/Geant4Particle.h
+++ b/DDG4/include/DDG4/Geant4Particle.h
@@ -118,7 +118,7 @@ namespace dd4hep {
       float spin[3]                { 0E0,0E0,0E0 };
       // 12 ints + 4 bytes + 3 floats should be aligned to 8 bytes....
       double vsx  = 0E0, vsy  = 0E0, vsz = 0E0;
-      double vex  = 0E0, vey  = 0E0, vez = 0E0;
+      double vex  = 0E0, vey  = 0E0, vez = 0E0, vet = 0E0;
       double psx  = 0E0, psy  = 0E0, psz = 0E0;
       double pex  = 0E0, pey  = 0E0, pez = 0E0;
       double mass = 0E0, time = 0E0, properTime = 0E0;
