diff --git a/source/processes/electromagnetic/utils/src/G4EmSaturation.cc b/source/processes/electromagnetic/utils/src/G4EmSaturation.cc
index 9fb850dd7..f47c56c02 100644
--- a/source/processes/electromagnetic/utils/src/G4EmSaturation.cc
+++ b/source/processes/electromagnetic/utils/src/G4EmSaturation.cc
@@ -269,7 +269,7 @@ void G4EmSaturation::DumpG4BirksCoefficients()
 
 void G4EmSaturation::InitialiseG4materials()
 {
-  nG4Birks = 4;
+  nG4Birks = 13;
   g4MatData.reserve(nG4Birks);
 
   // M.Hirschberg et al., IEEE Trans. Nuc. Sci. 39 (1992) 511
@@ -305,6 +305,30 @@ void G4EmSaturation::InitialiseG4materials()
   //G4_PbWO4 - CMS value
   g4MatNames.push_back("G4_PbWO4");
   g4MatData.push_back(0.0333333*mm/MeV);
+  // PbWO4 used in HybdriCalorimeter
+  g4MatNames.push_back("PbWO4");
+  g4MatData.push_back(0.0333333*mm/MeV);
+  // Estimate SciGlass to be similar to PbWO4 as it still has to be measured
+  g4MatNames.push_back("SciGlass");
+  g4MatData.push_back(0.0333333*mm/MeV);
+
+  // Various plastic scintillator materials
+  g4MatNames.push_back("PlasticScint079");
+  g4MatData.push_back(0.07943*mm/MeV);
+
+  g4MatNames.push_back("polystyrene");
+  g4MatData.push_back(0.126*mm/MeV);
+  g4MatNames.push_back("Polystyrene");
+  g4MatData.push_back(0.126*mm/MeV);
+  g4MatNames.push_back("PlasticScint");
+  g4MatData.push_back(0.126*mm/MeV);
+  g4MatNames.push_back("PlasticScint126");
+  g4MatData.push_back(0.126*mm/MeV);
+  g4MatNames.push_back("PlasticScintillator");
+  g4MatData.push_back(0.126*mm/MeV);
+
+  g4MatNames.push_back("PlasticScint153");
+  g4MatData.push_back(0.153*mm/MeV);
 
   //G4_Lucite
 
